cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -602,43 +603,111 @@ func (m *McmManager) retry(fn func(ctx context.Context) (bool, error), resourceT
 	}
 }
 
-// GetMachineDeploymentNodes returns the set of Nodes which belongs to the MachineDeployment.
-func (m *McmManager) GetMachineDeploymentNodes(machinedeployment *MachineDeployment) ([]string, error) {
-	md, err := m.machineDeploymentLister.MachineDeployments(m.namespace).Get(machinedeployment.Name)
-	if err != nil {
-		return nil, fmt.Errorf("Unable to fetch MachineDeployment object %s, Error: %v", machinedeployment.Name, err)
-	}
+// GetMachineDeploymentInstances returns list of cloudprovider.Instance for machines which belongs to the MachineDeployment.
+func (m *McmManager) GetMachineDeploymentInstances(machinedeployment *MachineDeployment) ([]cloudprovider.Instance, error) {
```

Comment:
`GetInstancesForMachineDeployment` sounds more clear to me compared to `GetMachineDeploymentInstances`.